### ISSUE TITLE
What is `bindkey -M vicmd "^[[3~" delete-char` for?

### ISSUE THREAD

**Issue by kutsan** [hlissner/dotfiles#2]
I was reading your dotfiles and came across these lines. I tried to search for it but couldn't find anything. Do you care to explain what these lines for? Actually, what is `^[[3~`? Is there any resource I can learn about them more?

```sh
bindkey -M vicmd "^[[3~" delete-char
bindkey "^[[3~"  delete-char
bindkey "^[3;5~" delete-char
```

**Comment by hlissner** [comment]
"^[[3~" is the escape code for the delete key. In zsh, this key doesn't delete-forward as I would expect. Instead, it acts as if I've pressed `~e` in command mode (or `g~e` in vim), which flips the case of the word after the cursor. 

Not sure if that's a quirk of zsh's vi mode, but I manually bound delete to `delete-char` to fix it. 

"^[3;5~" is actually a typo. It's supposed to be `^[[3;5~`, which is the escape code for ctrl + delete. There's also `^[[3;2~` for shift-delete. I don't think that one is necessary in either case though.

If you're wondering how I code those codes, I ran `cat`, then pressed the key I wanted, and it prints out the escape code, e.g.

```sh
$ cat
^[[3~
```

Hope that helps!

**Comment by kutsan** [comment]
So much thanks! You and `man {terminfo,infocmp,zshzle}` (those were not just for this topic) helped me a lot. Here is what I learned. As I understand, there is no quirk here. ZLE applies everything it receives. In this case it's `^[[3~`:

1. `^[` It's escape character, so switches to normal mode.
1. `[` Since it's `undefined-key`, so ZLE doesn't know what to do with.
1. `3~` Flips the next 3 letters.

I tested it with other `undefined-key`s like <kbd>FN + RightArrow</kbd>. `cat` outputs `^[[4~` for that key sequences. It exactly flips the next 4 letters like I said above.

> I ran cat, then pressed the key I wanted, and it prints out the escape code, e.g.

`cat` trick is nice but I assume we can achieve the same result with pressing `C-v` and hitting a key, or `sed -n l`.

Thanks again. <3
